### ISSUE TITLE
DeDRM not  working in Calibre 4.23

### ISSUE THREAD

**Issue by NeilH37** [apprenticeharper/DeDRM_tools#1439]
I have been successfully using DeDRM 6.8.0 and Calibre 4.23 for sometime (I reinstalled 4.23 a little while a go when similar problem arose following my updating Calbre, this solved the problem till now.
I have download an AZW3 file and added it to Calibre but it was still DRM protected.  I ran convert AZW3 to AZW3 - failed with <This book is locked by DRM ....>.  Tried again but this time to convert to ePub, same result.
Hope this post is OK, I am new to Git Hub as I arrived here by following links from Calibre Plugin Help.  Apologies if I have got it wrong.

- [ ] `The Title` and The `Log Title` are setted correctly.
- [ ] Clarified about `my environment`.
- [ ] Code block is used for `the log`.

---

## Log
<details><summary>Log Title</summary>

```log
PUT YOUR LOG
```
</details>

**Comment by ElleKayEm** [comment]
Let's see a log of adding the book to calibre:  https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md#i-cannot-solve-my-problem-with-the-dedrm-plugin-and-now-i-need-to-post-a-log-how-do-i-do-that

**Comment by NeilH37** [comment]
calibre Debug log
calibre 4.23  embedded-python: True is64bit: False
Windows-10-10.0.19041-SP0 Windows ('32bit', 'WindowsPE')
32bit process running on 64bit windows
('Windows', '10', '10.0.19041')
Python 2.7.16+
Windows: ('10', '10.0.19041', 'SP0', u'Multiprocessor Free')
Interface language: en_GB
Successfully initialized third party plugins: DeDRM (6, 8, 0) && Obok DeDRM (6, 7, 0) && Find Duplicates (1, 8, 3)
calibre 4.23  embedded-python: True is64bit: False
Windows-10-10.0.19041-SP0 Windows ('32bit', 'WindowsPE')
32bit process running on 64bit windows
('Windows', '10', '10.0.19041')
Python 2.7.16+
Windows: ('10', '10.0.19041', 'SP0', u'Multiprocessor Free')
Interface language: en_GB
Successfully initialized third party plugins: DeDRM (6, 8, 0) && Obok DeDRM (6, 7, 0) && Find Duplicates (1, 8, 3)
Turning on automatic hidpi scaling
devicePixelRatio: 1.0
logicalDpi: 96.0 x 96.0
physicalDpi: 100.861627907 x 100.55257732
Using calibre Qt style: True
[0.00] Starting up...
[0.00] Showing splash screen...
[0.55] splash screen shown
[0.55] Initializing db...
[0.56] db initialized
[0.56] Constructing main UI...
DEBUG:    0.0 obok::utilities.py - loading translations
DEBUG:    0.0 obok::dialogs.py - loading translations
DEBUG:    0.0 obok::config.py - loading translations
DEBUG:    0.0 obok::action_err.py - loading translations
Loaded cached store plugin for: Amazon Kindle at version: 19
Loaded cached store plugin for: ebook.de at version: 6
Loaded cached store plugin for: Project Gutenberg at version: 7
Loaded cached store plugin for: Google Books at version: 7
[2.72] main UI initialized...
[2.72] Hiding splash screen
[58.97] splash screen hidden
[58.97] Started up in 58.97 seconds with 272 books
stdout+stderr from file dialog helper: ['', '']
piped data from file dialog helper: ['\xb5M\x81\x1fh\xd7\xd2kb\x94\x93\x8e\xdf:\xfbG\xc8\x05\xc4\xbaV\xb6c\xb4\xcc\xb3\x88\x80\xfcv\xff\x15', 'C:\\Users\\Neil\\Desktop\\Documents\\Temp\\eBook Download Temp storage\\Care for Your Guinea Pigs (RSPCA Pet Guide).azw3']
DeDRM v6.8.0: Trying to decrypt Care for Your Guinea Pigs (RSPCA Pet Guide).azw3
Using Library AlfCrypto DLL/DYLIB/SO
MobiDeDrm v0.42.
Copyright © 2008-2017 The Dark Reverser, Apprentice Harper et al.
Decrypting Kindle Format 8 ebook: Care for Your Guinea Pigs (RSPCA Pet Guide)
Found 2 keys to try after 0.4 seconds
Crypto Type is: 2
DeDRM v6.8.0: Failed to decrypt with error: No key found in 2 keys tried.
DeDRM v6.8.0: Looking for new default Kindle Key after 0.4 seconds
searching for kinfoFiles in C:\Users\Neil\AppData\Local
Found K4PC 1.25+ kinf2018 file: C:\Users\Neil\AppData\Local\Amazon\Kindle\storage\.kinf2018
Decrypted key file using IDString '616134635' and UserName '4e65696c'
DeDRM v6.8.0: Found 1 new key
MobiDeDrm v0.42.
Copyright © 2008-2017 The Dark Reverser, Apprentice Harper et al.
Decrypting Kindle Format 8 ebook: Care for Your Guinea Pigs (RSPCA Pet Guide)
Got DSN key from database default_key_1
Found 4 keys to try after 11.2 seconds
Crypto Type is: 2
DeDRM v6.8.0: Ultimately failed to decrypt after 11.2 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 172, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 625, in run
  File "calibre_plugins.dedrm.__init__", line 575, in KindleMobiDecrypt
DeDRMError: DeDRM v6.8.0: Ultimately failed to decrypt after 11.2 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Added Care for Your Guinea Pigs (RSPCA Pet Guide) to db in: 2.8
Added 1 books in 15.2 seconds

**Comment by ElleKayEm** [comment]
This looks like a book from the website for "download and transfer via usb".  Did you chose an E-ink Kindle (not a Fire) for the device?  And have you entered the Kindle's serial number into DeDRM's customization?

**Comment by NeilH37** [comment]
My apologies and thanks, root cause of the problem was <idiot operator>.  I recently purchased a new Kindle Paperwhite but for some time had both registered with Amazon.  A couple of weeks ago, I gave my old Kindle to my granddaughter and deregistered it with Amazon, but I forgot all about DeDRM's holding the serial number.  Correct serial number now being used and all works well.  Thanks again, clear instructions and impressively short response time.
